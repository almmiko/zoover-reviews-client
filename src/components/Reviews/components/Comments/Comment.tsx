import React from 'react';
import { Wrapper, Header, RatingNumber, Title, TitleText, TitleSubInfo, RatingAspects, Aspect, AspectDescription, AspectRating, Description, MetaInfo } from './elements';

class Comment extends React.Component<any> {
  render() {
    return (
      <Wrapper>
        <Header>
          <RatingNumber>8</RatingNumber>
          <Title>
            <TitleText>Tittle</TitleText>
            <TitleSubInfo>user, 13-12-44</TitleSubInfo>
          </Title>
        </Header>
        <RatingAspects>
          <Aspect>
            <AspectDescription>Location</AspectDescription>
            <AspectRating>5</AspectRating>
          </Aspect>
          <Aspect>
            <AspectDescription>Location</AspectDescription>
            <AspectRating>5</AspectRating>
          </Aspect>
          <Aspect>
            <AspectDescription>Location</AspectDescription>
            <AspectRating>5</AspectRating>
          </Aspect>
          <Aspect>
            <AspectDescription>Location</AspectDescription>
            <AspectRating>5</AspectRating>
          </Aspect>
          <Aspect>
            <AspectDescription>Location</AspectDescription>
            <AspectRating>5</AspectRating>
          </Aspect>
          <Aspect>
            <AspectDescription>Location</AspectDescription>
            <AspectRating>5</AspectRating>
          </Aspect>
        </RatingAspects>
        <Description>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolorem doloremque earum incidunt quibusdam quod. Accusamus animi, aperiam est facilis iure odio provident totam voluptate. Dicta eligendi ex modi tempora tenetur!
        </Description>
        <MetaInfo>
          Traveled with family in 12.44.44
        </MetaInfo>
      </Wrapper>
    );
  }
}

export default Comment;
