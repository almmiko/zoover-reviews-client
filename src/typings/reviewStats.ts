export interface AverageRatingAspects {
  location: number;
  service: number;
  priceQuality: number;
  food: number;
  room: number;
  childFriendly: number;
  interior: number;
  size: number;
  activities: number;
  restaurants: number;
  sanitaryState: number;
  accessibility: number;
  nightlife: number;
  culture: number;
  surrounding: number;
  atmosphere: number;
  noviceSkiArea: number;
  advancedSkiArea: number;
  apresSki: number;
  beach: number;
  entertainment: number;
  environmental: number;
  pool: number;
  terrace: number;
}

export interface TraveledWith {
  FAMILY: number;
  FRIENDS: number;
  OTHER: number;
  COUPLE: number;
  SINGLE: number;
}

export interface ReviewStats {
  averageRating: number;
  averageRatingAspects: AverageRatingAspects;
  traveledWith: TraveledWith;
}
