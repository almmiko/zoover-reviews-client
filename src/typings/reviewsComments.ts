export interface Parent {
  id: string;
}

export interface General {
  general: number;
}

export interface Aspects {
  location: number;
  service: number;
  priceQuality: number;
  food: number;
  room: number;
  childFriendly: number;
  interior: number;
  size: number;
  activities: number;
  restaurants: number;
  sanitaryState: number;
  accessibility: number;
  nightlife: number;
  culture: number;
  surrounding: number;
  atmosphere: number;
  noviceSkiArea: number;
  advancedSkiArea: number;
  apresSki: number;
  beach: number;
  entertainment: number;
  environmental: number;
  pool: number;
  terrace: number;
  [key: string]: number;
}

export interface Ratings {
  general: General;
  aspects: Aspects;
}

export interface Titles {
  [key: string]: string;
}

export interface Texts {
  [key: string]: string;
}

export interface Resource {
  parents: Parent[];
  id: string;
  traveledWith: string;
  entryDate: number;
  travelDate: number;
  ratings: Ratings;
  titles: Titles;
  texts: Texts;
  user: string;
  locale: string;
}

export interface Meta {
  page: string;
  limit: string;
  totalPages: number;
}

export interface ReviewsComments {
  resources: Resource[];
  meta: Meta;
}
