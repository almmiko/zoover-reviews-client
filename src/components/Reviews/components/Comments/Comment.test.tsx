import { shallow } from 'enzyme';
import Comment from './Comment';
import React from 'react';
import { Resource } from '../../../../typings/reviewsComments';

const commentStub = {
  id: 'test',
  parents: [],
  traveledWith: 'test',
  entryDate: 1,
  travelDate: 1,
  user: 'test',
  locale: 'locale',
  texts: {
    locale: 'test'
  },
  titles: { locale: 'test' },
  ratings: { general: 12, aspects: [] } as any
};

it('renders correctly', () => {
  const tree = shallow(<Comment comment={commentStub as Resource} />);
  expect(tree).toMatchSnapshot();
});
