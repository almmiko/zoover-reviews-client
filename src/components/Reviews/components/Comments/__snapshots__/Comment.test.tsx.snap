// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div />
    <styled.div>
      <styled.h2>
        test
      </styled.h2>
      <styled.div>
        test
        , 
        01-01-1970
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div />
  <styled.div>
    test
  </styled.div>
  <styled.div>
    Traveled with 
    test
     in 
    January 1st 1970
  </styled.div>
</styled.div>
`;
